src/components/Link/Link.scss

```diff
@@ -18,6 +18,13 @@
   }
 }
 
+.IconLockup {
+  display: inline-flex;
+  align-items: center;
+
+  @include recolor-icon(currentColor);
+}
```

Comment:
In order to center the icon with the text, and to prevent the icon from wrapping by itself to the next line, the best solution I found is to wrap the last word and the icon in an inline-flex container.